cachi2/core/package_managers/gomod.py

```diff
@@ -82,7 +94,157 @@ class GoPackage(_GolangModel):
 
     import_path: str
     standard: bool = False
-    module: Optional[GoModule]
+    module: Optional[ParsedModule]
+
+
+class Module(NamedTuple):
+    """A Go module with relevant data for the SBOM generation"""
+
+    name: str
+    # name of the module before it was replaced
+    original_name: str
+    version: str
+    # The actual path to reach the module, differs from name in case of local replacements
+    resolved_path: str
+    main: bool = False
+
+    @property
+    def purl(self) -> str:
+        return f"pkg:golang/{self.resolved_path}@{self.version}?type=module"
```

Comment:
To make sure the purls are syntactically valid (percent-encoded where needed etc.), would it be worth using a library? https://github.com/package-url/packageurl-python

Reply:
I will add a sample commit using the library, so we can decide if the benefits make the decision worth it.